### PR TITLE
Create a exemplar Gradle Plugin

### PR DESCRIPTION
**Pull request by StefMa**
I've created a small Gradle Plugin for this project.
I think I've document everything enough to don't provide addional information in this PR description 😄 

The plugin is obviously *not* yet published to the portal.
But I've prepared some things for you.
Maybe we have to change/add something here as well. Like the license or so .. 🤷‍♂️

Beside of that I've added some questions. It would be great if someone can answer these 🙃 

I'm fine with it if you don't want - for whatever reasons - a plugin for this project.
Then give me a hint and I will move this to my personal github account 🙃

### CODE REVIEW

**Comment by eriwen** [comment]
Hi @StefMa. A plugin is definitely appropriate, and in fact I have a branch [ew/exemplar-gradle-plugin](https://github.com/gradle/exemplar/tree/ew/exemplar-gradle-plugin) that introduces and extension so you can configure where samples and docs are, and configure where the report should be output. 

I am happy to provide guidance here if you want to take that branch forward, as I probably won't be able to pursue a plugin for a little while. What do you think?

**Comment by StefMa** [comment]
Yeah, I'm happy to work on that plugin.
My next step would be to "merge" some of your "first steps for the plugin" into this branch.

> I am happy to provide guidance here 

Yes, please :)
A would appreciate a review here 👍

File: plugin/src/test/kotlin/org/gradle/exemplar/ExemplarPluginTest.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package org.gradle.exemplar
+
+import org.assertj.core.api.Assertions.assertThat
+import org.gradle.exemplar.internal.AbstractTest
+import org.junit.jupiter.api.Test
+
+class ExemplarPluginTest : AbstractTest {
```
[the remark this answers is not in the document]
Comment (StefMa) [reply plugin/src/test/kotlin/org/gradle/exemplar/ExemplarPluginTest.kt]: Oh wow. I wasn't aware of the `ProjectBuilder` ...

**Comment by StefMa** [comment]
> Even "just" setting up the repos and adding the relevant dependencies adds value and is a good first step.

I'm agree with you :)
Better to have multiple smaller PRs instead of one big one...

Anyway.
I've made the following changes:
* Always add *all* dependencies. Having two plugins feels a little bit wrong to me. 
* Uses the ProjectBuilder instead of the TestKit
* Updated the documentation

There are still open questions.
> // TODO: Do we really need java-library? Or is java-base enough?

Maybe @eriwen can answer this?

> // TODO: We have to inject the project-version into it and replace the hardcoded "0.6.1" how can we do that?

I tried it by making a task which injects the Project.version into a property file which can be then read by the plugin. But that feels wrong. The goal should be that as soon as we update the version in the "root build.gradle.kts" the version in the plugin will be changed too. 

> // FIXME: Hardcoding of the version is a bad idea.. Try to change this.. somehow.

How should we do that for the `org.slf4j:slf4j-simple` dependency? 🤔

**Comment by lacasseio** [comment]
There are some work we are currently doing as part of a "documentation" plugin suite that is expending off the guide plugins. There is a lot of overlap here and only should "survive". At the moment we are mostly struggling with the scope of the exemplar project and how it's used inside real life project. The current sample plugin use the exemplar project for testing. There are some oddity we are seeing as part of the usage for a more "general-purpose" sample documentation that we will address. Hold on a bit longer and we will hopefully have something more stable to share soon on this.

**Comment by StefMa** [comment]
If there is still interest in this, please let me know.

**Pull request closed without merging**